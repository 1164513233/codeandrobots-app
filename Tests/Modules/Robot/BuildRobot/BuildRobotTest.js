import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import BuildRobotScreen from 'App/Modules/Robot/BuildRobot/Screen'

describe('BuildRobot', () => {
  let wrapper

  const list = {
    stepOne: {
      image: null,
      title: '',
      text: ''
    },
    stepTwo: {
      image: null,
      title: '',
      text: ''
    },
    stepThree: {
      image: null,
      title: '',
      text: ''
    },
    stepFour: {
      image: null,
      title: '',
      text: ''
    }
  }

  beforeEach(() => {
    wrapper = shallow(
      <BuildRobotScreen
        list={list}
        navigation={{ navigate: () => { } }}
        showNotReadyModal={false}
        onHideNotReadyModal={() => { }}
        onPress={() => { }} />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
