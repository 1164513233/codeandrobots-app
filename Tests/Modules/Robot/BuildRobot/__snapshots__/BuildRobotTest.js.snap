// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildRobot Component renders correctly 1`] = `
<Container>
  <ScrollView
    style={
      Object {
        "marginBottom": 114,
      }
    }
  >
    <List
      linedRows={true}
      scrollable={true}
    >
      <CardListItem
        image={null}
        text=""
        title=""
      />
      <CardListItem
        image={null}
        text=""
        title=""
      />
      <CardListItem
        image={null}
        text=""
        title=""
      />
      <CardListItem
        image={null}
        text=""
        title=""
      />
    </List>
  </ScrollView>
  <Footer
    style={
      Object {
        "paddingTop": 34,
      }
    }
  >
    <Card
      button="Download Instructions"
    />
  </Footer>
</Container>
`;
